File: cpp/worker/src/common/env.cc
#include "gridmr/worker/common/env.h"
#include <cstdlib>

namespace gridmr_worker {

std::string envOr(const char* k, const char* d){
  const char* v = std::getenv(k);
  return (v && *v) ? std::string(v) : std::string(d);
}

} // namespace gridmr_worker

File: src/main/java/com/gridmr/master/util/Env.java
package com.gridmr.master.util;

public final class Env {
    private Env() {}

    public static String getEnvOrDefault(String key, String def) {
        String v = System.getenv(key);
        if (v == null || v.isEmpty()) return def;
        return v;
    }
}
